keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -53,17 +54,19 @@ class ArtifactListener(
       .forEach { artifact ->
         if (repository.isRegistered(artifact.name, artifact.artifactType)) {
           val artifactSupplier = artifactSuppliers.supporting(artifact.artifactType)
-          val version = artifactSupplier.getFullVersionString(artifact)
-          var status = artifactSupplier.getReleaseStatus(artifact)
-          log.info("Registering version {} ({}) of {} {}", version, status, artifact.name, artifact.type)
-          repository.storeArtifact(artifact.name, artifact.artifactType, version, status)
+          val enrichedArtifact = artifactSupplier.addMetadata(artifact.normalized())
```

Comment:
This method fetches metadata from the external CI service.

Reply:
fwiw, I wanted to implement it like this from the first place, but was concerned with external CI service failures :)